rooms.js

```diff
@@ -1147,15 +1147,22 @@ class BasicChatRoom extends BasicRoom {
 		let message = '';
 		if (this.introMessage) message += '\n|raw|<div class="infobox infobox-roomintro"><div' + (!this.isOfficial ? ' class="infobox-limited"' : '') + '>' + this.introMessage.replace(/\n/g, '') + '</div>';
 		if (this.staffMessage && user.can('mute', null, this)) message += (message ? '<br />' : '\n|raw|<div class="infobox">') + '(Staff intro:)<br /><div>' + this.staffMessage.replace(/\n/g, '') + '</div>';
+		if (message) message += '</div>';
+		return message;
+	}
+	/**
+	 * @param {User} user
+	 */
+	getRoomSettings(user) {
```

Comment:
Would make this a part of `getIntroMessage` instead of giving it its own function.